### PR TITLE
P1 Decal tests for null renderer

### PR DESCRIPTION
**Pull request by smurly**
Adds getter/setter P1 tests for Decal component. Sort Key is set to both 255 and 0 which are the boundary values of that property.

Adds remove method to component for simpler component remove. you can now just call dot remove() on a component object like a delete. Compliments the EditorEntity remove_component.

[SUCCESS] Success: Decal Entity successfully created
[SUCCESS] Success: Entity has a Decal component
[SUCCESS] Success: UNDO Entity creation success
[SUCCESS] Success: REDO Entity creation success
[SUCCESS] Success: Material property set on Decal component
[SUCCESS] Success: Attenuation Angle property set on Decal component
[SUCCESS] Success: Opacity property set on Decal component
[SUCCESS] Success: Sort Key property set on Decal component
[SUCCESS] Success: Sort Key property set on Decal component
[SUCCESS] Success: Decal component removed
[SUCCESS] Success: Entity has a Decal component
[SUCCESS] Success: Entered game mode
[SUCCESS] Success: Exited game mode
[SUCCESS] Success: Entity is hidden
[SUCCESS] Success: Entity is visible
[SUCCESS] Success: Entity deleted
[SUCCESS] Success: UNDO deletion success
[SUCCESS] Success: REDO deletion success
Test result:  SUCCESS

.\python\python.cmd -m pytest --build-directory atom_dev/bin/profile  .\AutomatedTesting\Gem\PythonTests\Atom\TestSuite_Main.py
====================================== test session starts
platform win32 -- Python 3.7.10, pytest-5.3.2, py-1.9.0, pluggy-0.13.1
rootdir: D:\workspace\o3de, inifile: pytest.ini
plugins: mock-2.0.0, timeout-1.3.4, ly-test-tools-1.0.0
collected 29 items

AutomatedTesting\Gem\PythonTests\Atom\TestSuite_Main.py .............................. [100%]

================================= 30 passed in 115.25s (0:01:55)

Signed-off-by: Scott Murray <scottmur@amazon.com>

### CODE REVIEW

**Review by SWMasterson — APPROVED** [review]

File: AutomatedTesting/Materials/decal/scorch_01_decal.material [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
```
Comment (jromnoa) [inline AutomatedTesting/Materials/decal/scorch_01_decal.material:1]: where is this material used?

File: AutomatedTesting/Materials/decal/airship_symbol_decal.material [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
```
Comment (jromnoa) [inline AutomatedTesting/Materials/decal/airship_symbol_decal.material:1]: same here, its not clear to me where this material is used

**Review by jromnoa — CHANGES_REQUESTED** [review]

File: AutomatedTesting/Materials/decal/scorch_01_decal.material (hunk 1 above)
Comment (smurly) [reply AutomatedTesting/Materials/decal/scorch_01_decal.material]: removed

File: AutomatedTesting/Materials/decal/airship_symbol_decal.material (hunk 2 above)
Comment (smurly) [reply AutomatedTesting/Materials/decal/airship_symbol_decal.material:1]: now set as material instead of default_gray

**Review by jromnoa — APPROVED** [review]
